### DISCUSSION TITLE
Transformations: Grouping to matrix empty value option

### DISCUSSION THREAD

**Discussion by hugo082** [grafana/grafana#55791] [Dashboards, Panels & Visualizations]
Currently, the value where there is an empty value in the input matrix is an empty string: `''`. But with this value, the output matrix is incompatible with some visualization options (e.g. _Connect null values_ on _State Timeline_ visualizer).

I would love to be able to use `null` as empty value to fix the _State Timeline_ visualizer. To avoid breaking changes, I suggest adding a new option to select the empty value with _Empty string_ as the default value.

I already open a PR, you can find it [here](https://github.com/grafana/grafana/pull/55591).

### Example

**Input data:**
| Time | Station | State |
|--|--|--|
|12:50|A|1|
|12:30|B|1|
|12:10|A|1|

**Output:**
| | A | B |
|--|--|--|
|12:50|1| |
|12:30| |1|
|12:10|1| |

**Expected:** (with `emptyValue = SpecialValue.Null`)
| | A | B |
|--|--|--|
|12:50|1| null |
|12:30| null |1|
|12:10|1| null |

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
